packages/sdk/src/utils/tryLoadEnvs.ts

```diff
@@ -84,34 +84,46 @@ function checkForConflicts(
   if (parsedRootEnv && envPath && areNotTheSame && fs.existsSync(envPath)) {
     const envConfig = dotenv.parse(fs.readFileSync(envPath))
     const conflicts: string[] = []
+    const duplicates: string[] = []
+
     for (const k in envConfig) {
-      if (parsedRootEnv[k] === envConfig[k]) {
-        conflicts.push(k)
+      const a = parsedRootEnv[k]
+      const b = envConfig[k]
+
+      if (a && b) {
+        if (a !== b) {
+          conflicts.push(k)
+        }
+
+        if (a === b) {
+          duplicates.push(k)
+        }
       }
     }
-    if (conflicts.length > 0) {
-      // const message = `You are trying to load env variables which are already present in your project root .env
-      const relativeRootEnvPath = path.relative(process.cwd(), rootEnvInfo!.path)
-      const relativeEnvPath = path.relative(process.cwd(), envPath)
-      if (type === 'error') {
-        const message = `There is a conflict between env var${conflicts.length > 1 ? 's' : ''} in ${chalk.underline(
-          relativeRootEnvPath,
-        )} and ${chalk.underline(relativeEnvPath)}
+
+    // const message = `You are trying to load env variables which are already present in your project root .env
+    const relativeRootEnvPath = path.relative(process.cwd(), rootEnvInfo!.path)
+    const relativeEnvPath = path.relative(process.cwd(), envPath)
+    if (type === 'error' && conflicts.length) {
+      const message = `There is a conflict between env var${conflicts.length > 1 ? 's' : ''} in ${chalk.underline(
+        relativeRootEnvPath,
+      )} and ${chalk.underline(relativeEnvPath)}
 Conflicting env vars:
 ${conflicts.map((conflict) => `  ${chalk.bold(conflict)}`).join('\n')}
 
 We suggest to move the contents of ${chalk.underline(relativeEnvPath)} to ${chalk.underline(
-          relativeRootEnvPath,
-        )} to consolidate your env vars.\n`
-        throw new Error(message)
-      } else if (type === 'warn') {
-        const message = `Conflict for env var${conflicts.length > 1 ? 's' : ''} ${conflicts
-          .map((c) => chalk.bold(c))
-          .join(', ')} in ${chalk.underline(relativeRootEnvPath)} and ${chalk.underline(relativeEnvPath)}
+        relativeRootEnvPath,
+      )} to consolidate your env vars.\n`
+      throw new Error(message)
+    }
+
+    if (type === 'warn' && duplicates.length) {
```

Comment:
What's the reason for preserving the old behavior when `type === 'warn'`?

My understanding was that the only thing that needed to change was the condition on line 88 (`if (parsedRootEnv[k] !== envConfig[k]) {` instead of `if (parsedRootEnv[k] === envConfig[k]) {`). Am I missing something?

Reply:
Oh, I missed the [comment](https://github.com/prisma/prisma/pull/12984#issuecomment-1110832049) about this.

I'm starting to have a feeling that it might be a lot more complicated that it seemed, and it would be a good topic for a TS Codebase Deep Dive meeting